### ISSUE TITLE
NoCheatPlus conflict with Magic

### ISSUE THREAD

**Issue by Blamo27** [NoCheatPlus/Issues#577]
Hello,

We're using the plugin Magic, that adds, amongst others, a spell called "glide" that allows you to glide like you would with elytrae, but without having some.

NCP survival fly protection prevents players from gliding using that spell. That is odd, considering that, on the other hand, players can use a spell called "fly" that allows them to fly like in creative.

Do you think you could work on a fix / an exception for that spell ? Thank you.

#### Complete output of the _ncp version_ command.
3.16.0-RC-sMD5NET-b1134

#### How to reproduce the issue ?
Cast the "fly" spell of the Magic plugin on a player.

#### Do you run any other special plugins that affect game mechanics next to NC+ ?
Magic in this case. (https://www.spigotmc.org/resources/magic.1056/)

#### Provide additional information, such as error/debug/violation logs.
NoCheatPlus moving violation.

**Comment by 0-x-2-2** [comment]
The creator of the plugin should fix it.

**Comment by NathanWolf** [comment]
Will adding a timed flight exemption be enough to chill NCP?

**Comment by 0-x-2-2** [comment]
NCP does have a api for plugins to use I think?

**Comment by NathanWolf** [comment]
Yes, and I use it, I just didn't think I had to add flight exceptions for vanilla gliding.

I've added one now, I guess I'll wait and see if that fixes the problem.

**Comment by Blamo27** [comment]
@NathanWolf @0-x-2-2 

NCP is still preventing user that is trying to use this spell. As you can see in the screen below :

![image](https://user-images.githubusercontent.com/971712/48666858-0bf99700-eaca-11e8-864a-38198e603004.png)

**Comment by NathanWolf** [comment]
I added a flight exemption. If NCP is tagging vanilla gliding as “creative flight” I’m not really going to do anything else about that.

**Comment by 0-x-2-2** [comment]
You are only doing survival_fly exemption if a player has vanilla flight enabled it will use the creative_fly check instead.

**Comment by NathanWolf** [comment]
I will try that, but to be clear I think the problem is gliding, not flying.

**Comment by 0-x-2-2** [comment]
Creative_fly is the check used for Elytra's too I think?

**Comment by NathanWolf** [comment]
Ok thanks, we'll give it a try!

@Blamo27 - unfortunately I'm out of town for the holidays and my build server seems to have gone down. So I made the fix, but it may be a week or two before you can get it.

**Comment by Blamo27** [comment]
@NathanWolf I can build it and test it if you want

**Comment by NathanWolf** [comment]
Sure that'd be great- the fix is pushed. I'm just adding exemptions for both creative and survival fly now.

**Comment by Blamo27** [comment]
Okey nice, I will test it tomorrow !
